### PR TITLE
properly handle completer

### PR DESCRIPTION
**Pull request by dshukertjr**
## What kind of change does this PR introduce?

Completers are used in gotrue client to be able to retry token refresh and return the final retry result, but currently the result of `_callRefreshToken` are directly being returned. 

This PR makes `_callRefreshToken` void and makes sure that the result of the completer are returned for those that needs to refresh the session.

### CODE REVIEW

File: lib/src/gotrue_client.dart [hunk 1]
```diff
@@ -643,7 +645,7 @@ class GoTrueClient {
   }
 
   /// Generates a new JWT.
-  Future<AuthResponse> _callRefreshToken(
+  Future<void> _callRefreshToken(
```
Comment (Vinzent03) [inline lib/src/gotrue_client.dart:648]: Can't this be changed to just `void`?

Comment (dshukertjr) [reply lib/src/gotrue_client.dart]: Yup, great catch!

**Comment by Vinzent03** [comment]
When using this pr as gotrue version in supabase-flutter, the test still fails... 🤔

**Review by Vinzent03 — APPROVED** [review]
This should be good. The supabase-flutter test fails because of [this line](https://github.com/supabase/gotrue-dart/blob/39f65a899fb1c3b6605ebaefb8913e84540434a3/lib/src/gotrue_client.dart#L699). When removing it, the test passes. So this pr should be fine.

**Comment by dshukertjr** [comment]
@Vinzent03 Thank for quickly identifying the cause of the failing tests! Yeah, this PR is kind of unrelated, but wanted to fix it. I think we want that addError there though, right?

**Comment by Vinzent03** [comment]
@dshukertjr Found the issue https://github.com/supabase/supabase-dart/pull/187

**Comment by dshukertjr** [comment]
@Vinzent03 ah, sorry, missed that one! You're awesome!

**Comment by Vinzent03** [comment]
You didn't miss it, it's brand new xD

**Pull request merged**